pkg/gas/block_history_estimator.go

```
@@ -255,6 +253,27 @@ func (b *BlockHistoryEstimator) GetLegacyGas(_ context.Context, _ []byte, gasLim
 	return
 }
 
+func (b *BlockHistoryEstimator) GetMaxLegacyGas(_ context.Context, _ []byte, _ uint64, maxGasPriceWei *assets.Wei, _ ...fees.Opt) (gasPrice *assets.Wei, chainSpecificGasLimit uint64, err error) {
+	if b.eConfig.EIP1559DynamicFees() {
+		return gasPrice, 0, errors.New("can't get max legacy gas, EIP1559 is enabled")
+	}
+
+	ok := b.IfStarted(func() {
+		gasPrice = b.getMaxPercentileGasPrice()
+		if gasPrice == nil {
+			err = errors.New("BlockHistoryEstimator has not finished the first gas estimation yet")
+			return
+		}
+	})
```

This works ok, but the `return` here will only return from the inner func, which is iffy: the `err` might be set but it's not checked. Maybe clearer to add an `if err != nil` on line 268 or 270

---

Yeah I think there is a silent error here . If there is an error, "ok" will still be true and we'll go to line 274

---

(And on line 274 we'd return "nil" for gas, which seems incorrect)